tensilelite/Tensile/KernelWriterAssembly.py

```diff
@@ -2005,6 +2016,15 @@ def graWorkGroup(self, kernel):
     ########################################
     # Blocked rows or columns
     # Do branch
+    
+    # Restore WGM
+    tmpSgpr = self.sgprPool.checkOut(2, preventOverflow=0)
+    module.add(SOrB32(dst=sgpr(tmpSgpr), src0=sgpr("WGM"), src1=hex(0xFFFFF000), comment=""))
+    module.add(SAndB32(dst=sgpr("WGM"), src0=sgpr("WGM"), src1=hex(0x00000FFF), comment=""))
+    module.add(SAndB32(dst=sgpr(tmpSgpr+1), src0=sgpr("WGM"), src1=hex(0x800), comment=""))
+    module.add(SCSelectB32(dst=sgpr("WGM"), src0=sgpr(tmpSgpr), src1=sgpr("WGM"), comment=""))
+    self.sgprPool.checkIn(tmpSgpr)
```

Comment:
WGM has negative values, can we keep it 16-bit so we can use native instruction to convert I16 to I32?

Reply:
Is it appropriate to set the range for XCC to be range(0, 128) and XCCG in range(1, 512), seems like we do not exceed these ranges?

Reply:
updated. 
add SSExtI16toI32